### DISCUSSION TITLE
ScriptReference/AssetImporters.ScriptedImporterAttribute-ctor

### DISCUSSION THREAD

**Discussion by giscus** [Developer-Notes-Extension/Unity-Notes#527] [Notes]
# ScriptReference/AssetImporters.ScriptedImporterAttribute-ctor

https://docs.unity3d.com/ScriptReference/AssetImporters.ScriptedImporterAttribute-ctor.html

**Comment by radiatoryang** [comment]
import order offsets of built-in Unity importers from https://forum.unity.com/threads/understanding-import-order-of-native-unity-asset-types.1187845/
Texture2D: -2004/-2003
ScriptableObject: -801/-800
Prefab: 500/501

**Reply by radiatoryang** [reply]
also, chaining together different imports / doing multi-pass importing is complicated... see https://forum.unity.com/threads/create-additional-prefab-assets-in-scriptedimporter-and-track-them.1158734/

> 1:
Import the ScriptedImporter. It adds a basic root GameObject to the Hierarchy, essentially a blank object at first.
The reason we add a basic GameObject is so that the AssetImportContext.SetMainObject can be fulfilled for the first import to not cause potential import issues.
(At this point in time, Prefabs are NOT imported yet; they would be null.)
2:
Do the EditorApplication.delayCall to reimport this asset slightly later, so that prefabs are imported and ready to be used in the actual import process.
3:
The EditorApplication.delayCall will now invoke the second reimport.
The prefabs are now available to load and do not cause any issues that were brought up in the two situations discussed above.
